state-sync/storage-service/types/src/responses.rs

```diff
@@ -469,6 +477,14 @@ impl DataSummary {
             }
         }
     }
+
+    /// Returns true iff the optimistic data request can be serviced
+    fn can_service_optimistic_request(&self, known_version: u64) -> bool {
+        self.synced_ledger_info
+            .as_ref()
+            .map(|li| (li.ledger_info().version() + OPTIMISTIC_FETCH_VERSION_DELTA) > known_version)
```

Comment:
hmm, isn't this meaning that the remote data needs to be OPTIMISTIC_FETCH_VERSION_DELTA more than the request version in order to serve it?

Reply:
oh so this is saying that if the upstream is not OPTIMISTIC_FETCH_VERSION_DELTA behind us, we can do the sync